### ISSUE TITLE
@babel/plugin-transform-regenerator@7.4.5 causing unnecessary variable renaming

### ISSUE THREAD

**Issue by ArgonAlex** [babel/babel#10094]
## Bug Report

**Current Behavior**
@babel/plugin-transform-regenerator@7.4.5 seems to be causing @babel/plugin-transform-block-scoping to unnecessarily rename variables in scopes after a generator function.

```js
var _marked =
/*#__PURE__*/
regeneratorRuntime.mark(f);

function f() {
  var a;
  return regeneratorRuntime.wrap(function f$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          a = 0;

        case 1:
        case "end":
          return _context.stop();
      }
    }
  }, _marked);
}

if (true) {
  var _a; // HERE

}
```

**Input Code**
```js
function* f() {
  let a = 0;
}
if (true) {
  let a; // HERE
}
```

**Expected behavior/code**
WIth @babel/plugin-transform-regenerator@7.4.4
```js
var _marked =
/*#__PURE__*/
regeneratorRuntime.mark(f);

function f() {
  var a;
  return regeneratorRuntime.wrap(function f$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          a = 0;

        case 1:
        case "end":
          return _context.stop();
      }
    }
  }, _marked);
}

if (true) {
  var a; // HERE
}
```

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  plugins: [
    "transform-block-scoping",
    "regenerator"
  ]
}
```

**Environment**
- Babel version(s): 7.4.5

**Comment by ArgonAlex** [comment]
@babel-bot Move to facebook/regenerator

**Comment by JLHwung** [comment]
I can not reproduce this issue on REPL: https://babel.dev/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mAVAAmACgJTIN4ChnIA2AplMgIbIC8yADANy4C-uMwyaUATiMVngRJly9ZAHoxyABIBRAEozmQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ces2015%2Cenv&prettier=true&targets=&version=7.10.5&externalPlugins=

Feel free to reopen it if you can come up with a reproducing REPL example.

**Comment by ArgonAlex** [comment]
Nice, it looks like it was fixed in either 7.6.1 or 7.6.2

7.6.0 Broken:
https://babel.dev/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mAVAAmACgJTIN4ChnIA2AplMgIbIC8yADANy4C-uMwyaUATiMVngRJly9ZAHoxyABIBRAEozmQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Cenv&prettier=true&targets=&version=7.6.0&externalPlugins=

7.6.2 Fixed:
https://babel.dev/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mAVAAmACgJTIN4ChnIA2AplMgIbIC8yADANy4C-uMwyaUATiMVngRJly9ZAHoxyABIBRAEozmQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Cenv&prettier=true&targets=&version=7.6.2&externalPlugins=
